core/src/main/scala/codesearch/core/search/SearchRequest.scala

```diff
@@ -12,35 +15,35 @@ package codesearch.core.search
 case class SearchRequest(
     query: String,
     filter: Option[String],
+    filePath: Option[String],
     insensitive: Boolean,
     spaceInsensitive: Boolean,
     preciseMatch: Boolean,
     sourcesOnly: Boolean,
     page: Int,
-    callURI: String
+    callURI: Uri
```

Comment:
I don't think `callURI` should be a part of the request. Make a special method in SearchRequest with `host` parameter for creating `callUri`.